File: frontend/src/services/scheduledJob.ts
import axios from 'axios';

const api = axios.create({
  baseURL: '/api/v1',
  timeout: 30000,
});

// 类型定义
export interface ScheduledJob {
  id: number;
  name: string;
  description?: string;
  job_type: 'testcase' | 'suite';
  target_id: number;
  target_name?: string;
  cron_expression: string;
  device_serial?: string;
  enabled: boolean;
  status: 'pending' | 'running' | 'completed' | 'failed' | 'disabled';
  last_run_time?: string;
  next_run_time?: string;
  last_run_status?: string;
  last_run_message?: string;
  project_id?: number;
  created_by?: number;
  created_at: string;
  updated_at: string;
}

export interface ScheduledJobCreate {
  name: string;
  description?: string;
  job_type: 'testcase' | 'suite';
  target_id: number;
  cron_expression: string;
  device_serial?: string;
  enabled?: boolean;
  project_id?: number;
}

export interface ScheduledJobUpdate {
  name?: string;
  description?: string;
  job_type?: 'testcase' | 'suite';
  target_id?: number;
  cron_expression?: string;
  device_serial?: string;
  enabled?: boolean;
  project_id?: number;
}

// 获取定时任务列表
export const getScheduledJobs = (params?: {
  page?: number;
  page_size?: number;
  enabled_only?: boolean;
  project_id?: number;
}) => {
  return api.get('/scheduled-jobs', { params });
};

// 获取定时任务详情
export const getScheduledJob = (id: number) => {
  return api.get(`/scheduled-jobs/${id}`);
};

// 创建定时任务
export const createScheduledJob = (data: ScheduledJobCreate) => {
  return api.post('/scheduled-jobs', data);
};

// 更新定时任务
export const updateScheduledJob = (id: number, data: ScheduledJobUpdate) => {
  return api.put(`/scheduled-jobs/${id}`, data);
};

// 删除定时任务
export const deleteScheduledJob = (id: number) => {
  return api.delete(`/scheduled-jobs/${id}`);
};

// 手动运行定时任务
export const runScheduledJob = (id: number, deviceSerial: string) => {
  return api.post(`/scheduled-jobs/${id}/run`, { device_serial: deviceSerial });
};

// 切换定时任务启用状态
export const toggleScheduledJob = (id: number) => {
  return api.post(`/scheduled-jobs/${id}/toggle`);
};

File: backend/app/models/scheduled_job.py
"""
Scheduled Job model - automated test execution
"""
from sqlalchemy import Column, String, Text, Integer, Boolean, DateTime, ForeignKey, Enum as SQLEnum
from sqlalchemy.orm import relationship
from .base import BaseModel, Base
import enum


class JobStatus(enum.Enum):
    """Scheduled job status"""
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    DISABLED = "disabled"


class ScheduledJob(BaseModel):
    """Scheduled Job model for automated test execution"""
    __tablename__ = 'scheduled_jobs'

    name = Column(String(200), nullable=False, unique=True, comment='Job name')
    description = Column(Text, nullable=True, comment='Job description')
    job_type = Column(String(50), nullable=False, comment='Job type: testcase, suite')
    target_id = Column(Integer, nullable=False, comment='Target testcase or suite ID')
    cron_expression = Column(String(100), nullable=False, comment='Cron expression')
    device_serial = Column(String(100), nullable=True, comment='Device serial for execution')
    enabled = Column(Boolean, nullable=False, default=True, comment='Job enabled flag')
    status = Column(SQLEnum(JobStatus), nullable=False, default=JobStatus.PENDING, comment='Job status')
    last_run_time = Column(DateTime, nullable=True, comment='Last execution time')
    next_run_time = Column(DateTime, nullable=True, comment='Next scheduled run time')
    last_run_status = Column(String(20), nullable=True, comment='Last run status: success, failed')
    last_run_message = Column(Text, nullable=True, comment='Last run message')
    project_id = Column(Integer, ForeignKey('projects.id', ondelete='SET NULL'), nullable=True, comment='Project ID')
    created_by = Column(Integer, nullable=True, comment='Creator user ID')

    # Relationships
    project = relationship('Project')

    def __repr__(self):
        return f"<ScheduledJob(id={self.id}, name='{self.name}', type='{self.job_type}', cron='{self.cron_expression}')>"
